packages/api/src/routes/studies.ts

```diff
@@ -37,13 +37,21 @@ router.delete("/:id", isAuthenticated, async (req, res, next) => {
 });
 
 router.get("/:id", isAuthenticated, (req, res, next) => {
-  Study.findById(req.params["id"])
-    .populate<{ batteries: ICustomizedBattery[] }>("batteries")
-    .then((study) => {
-      if (!study) return next(new HttpError(404));
-      res.json(study);
-    })
-    .catch(next);
+  try {
+    const user = req.user as HydratedDocument<IUser>;
+
+    Study.findById(req.params["id"])
+      .populate<{ batteries: ICustomizedBattery[] }>("batteries")
+      .then((study) => {
+        if (study?.owner !== user) {
+          return next(new HttpError(404));
+        }
+        res.json(study);
+      })
+      .catch(next);
+  } catch (e) {
+    next(e);
+  }
 });
 
 router.post("/new", isAuthenticated, async (req, res, next) => {
```

Comment:
can't leave a comment on unchanged code but we'll need to set owner to user._id on line 60 here